### ISSUE TITLE
Inconsistent location of IntAct xrefs

### ISSUE THREAD

**Issue by krchristie** [geneontology/go-ontology#14272]
While putting in a new complex term for @bmeldal, I noticed that there is some inconsistency with the location of the xrefs to IntAct IDs. There are over 200 hits for the string "IntAct:EBI", sometimes more than 1 for a given GO term.

I would have thought that this type of reference should be a general xrefs, but of the small subset I looked at, the majority are definition xrefs, instead of general xrefs:

**Terms with "IntAct:EBI" in a definition dbxref:**
- ANPR-A receptor complex
- ATF1-ATF4 transcription factor complex (2)
- ATF4-CREB1 transcription factor complex (2)
- ATG1/ULK1 kinase complex
- BID-BCL-xl complex (3)
- BRE1 E3 ubiquitin ligase complex
- Beta-catenin-ICAT complex (2)

**Terms with "IntAct:EBI" in a database_cross_reference:**
- B-WICH complex

I'm fine with whichever location is preferred, but it seems they should all be in the same one.

**Comment by bmeldal** [comment]
There should really be nearly 2000 xrefs as almost all complexes in the CP are linked to a complex term in GO. 

>1 xrefs to the same term are normal as we annotate different species that have the same functional complexes.

Generally, they are updated with every GOA release from a file we supply from our release. But I also know there are some issues with ECO terms that make our annotations stuck in the P2GO pipeline and we are looking at the solutions to it right now (not an easy job as it has several component that need fixing). But those annotations are mainly MFs and BPs and shouldn't affect the integration of the CP xrefs to the GO complex terms.

**Comment by krchristie** [comment]
Thanks @bmeldal for adding info about why it is correct to have more than IntAct xref to a single GO term. I wasn't trying to suggest that it wasn't, just trying to point out that the number of IntAct references didn't correspond 1:1 to the number of GO terms.

Looking directly at the file, this is what I see:
- 184 references to 171 IntAct xrefs in definition dbxrefs for 108 GO terms
- 13 IntAct xrefs as general xrefs for 7 GO terms (Note that for two of these GO terms, 4 IntAct IDs are present BOTH as definition dbxrefs and as general xrefs)

I also noticed that some IntAct IDs are used for more than one GO term. For EC number xrefs, I think we normally try to only attach the EC ID to the single term it corresponds to, so this seems weird to me. Here are the cases I found where the same IntAct ID is attached to more than one current (i.e. not obsolete) term. Does it seem right to you for these IDs to be attached to more than one GO term?

**IntAct:EBI-10098921 & IntAct:EBI-10099454** (these two IDs both present in two terms)

> [Term]
> id: GO:1990661
> name: S100A8 complex
> namespace: cellular_component
> def: "A protein complex composed of a S100A8 dimer and capable of binding to toll-like receptor 4 (TLR4)." [GOC:bhm, IntAct:EBI-10098921, IntAct:EBI-10099454, PMID:25417112]
> created_by: bhm
> creation_date: 2015-02-25T15:33:32Z
> 
> [Term]
> id: GO:1990662
> name: S100A9 complex
> namespace: cellular_component
> def: "A protein complex composed of a S100A9 dimer and capable of binding to toll-like receptor 4 (TLR4) and the receptor for advanced glycation end products (RAGE) initiating signal transduction through NF-kappa-B pathways. Transports arachidonic acid between the cytosol and the NADPH oxidase complex at the plasma membrane in neutrophils as part of an inflammatory signal cascade leading to an oxidative burst. Complexes with microtubules to increase cell motility." [GOC:bhm, IntAct:EBI-10098921, IntAct:EBI-10099454, PMID:15642721]
> created_by: bhm
> creation_date: 2015-02-25T15:55:48Z


**IntAct:EBI-10105915 & IntAct:EBI-10106087** (these two IDs both present in two terms)

> [Term]
> id: GO:1990657
> name: iNOS-S100A8/A9 complex
> namespace: cellular_component
> def: "A protein complex capable of stimulus-inducible nitric-oxide synthase activity. S-nitrosylates cysteine residues in target proteins, a principal mechanism of nitric oxide (NO)-mediated signal transduction. In mammals consists of NOS2, S100A8 and S100A9. S100A9 acts both as an adaptor linking NOS2 to its target and as a tra\
> nsnitrosylase that transfers the nitric oxide moiety from NOS2 to its target, via its own S-nitrosylated cysteine." [GOC:bhm, IntAct:EBI-10105915, IntAct:EBI-10106087, PMID:25417112]
> comment: An example of this is NOS2 in human (UniProt symbol P35228) in PMID:25417112 (inferred from direct assay).
> is_a: GO:1903958 ! nitric-oxide synthase complex
> is_a: GO:1990658 ! transnitrosylase complex
> relationship: capable_of GO:0035605 ! peptidyl-cysteine S-nitrosylase activity
> relationship: capable_of_part_of GO:0018119 ! peptidyl-cysteine S-nitrosylation
> relationship: part_of GO:0005737 ! cytoplasm
> created_by: bhm
> creation_date: 2015-02-25T11:00:49Z
> 
> [Term]
> id: GO:1990658
> name: transnitrosylase complex
> namespace: cellular_component
> def: "A transferase complex which is capable of transferring nitrogenous groups from one component to another." [GOC:bhm, IntAct:EBI-10105915, IntAct:EBI-10106087, PMID:25417112]
> comment: An example of this is S100A9 in human (UniProt symbol P06702) in PMID:25417112 (inferred from direct assay).
> synonym: "transferase complex, transferring nitrogenous groups" EXACT []
> is_a: GO:1990234 ! transferase complex
> created_by: bhm
> creation_date: 2015-02-25T11:17:17Z


**IntAct:EBI-959455** (present in two terms, once as sole IntAct xref, once as one of three xrefs)

> [Term]
> id: GO:1990350
> name: glucose transporter complex
> namespace: cellular_component
> def: "A protein complex facilitating glucose transport into, out of or within a cell, or between cells." [GOC:bhm, IntAct:EBI-959455, IntAct:EBI-960644, IntAct:EBI-960674, PMID:15449578]
> comment: An example of this is GTR1 in human (UniProt symbol P11166) in PMID:15449578 (inferred from direct assay).
> is_a: GO:1990351 ! transporter complex
> created_by: bhm
> creation_date: 2014-03-26T14:12:05Z
> 
> [Term]
> id: GO:1990351
> name: transporter complex
> namespace: cellular_component
> def: "A protein complex facilitating transport of molecules (proteins, small molecules, nucleic acids) into, out of or within a cell, or between cells." [GOC:bhm, IntAct:EBI-959455, PMID:15449578]
> comment: An example of this is GTR1 in human (UniProt symbol P11166) in PMID:15449578 (inferred from direct assay).
> intersection_of: GO:0032991 ! macromolecular complex
> intersection_of: capable_of GO:0005215 ! transporter activity
> created_by: bhm
> creation_date: 2014-03-26T14:23:27Z


**IntAct:EBI-9985210 & IntAct:EBI-9985797** (these two IDs both present in two terms)

> [Term]
> id: GO:1990562
> name: syndecan-syntenin-ALIX complex
> namespace: cellular_component
> def: "An exosome complex that is assembled in the multivesicular body (MVB) membrane and chaperoned to the exosome by the ESCRT-III machinery." [GOC:bhm, IntAct:EBI-9985210, IntAct:EBI-9985797, PMID:22660413]
> comment: An example of this is SCD1 in human (UniProt symbol P18827) in PMID:22660413 (inferred from physical interaction).
> synonym: "exosome complex" BROAD []
> is_a: GO:1990563 ! extracellular exosome complex
> created_by: bhm
> creation_date: 2014-11-25T16:30:47Z
> 
> [Term]
> id: GO:1990563
> name: extracellular exosome complex
> namespace: cellular_component
> def: "A protein complex that is wholly or partially contained within the lumen or membrane of the extracellular vesicular exosome." [GOC:bhm, IntAct:EBI-9985210, IntAct:EBI-9985797, PMID:22660413]
> comment: An example of this is SCD1 in human (UniProt symbol P18827) in PMID:22660413 (inferred from physical interaction).
> synonym: "exosome complex" EXACT []
> synonym: "extracellular vesicular exosome complex" EXACT [GOC:vesicles]
> is_a: GO:0043234 ! protein complex
> created_by: bhm
> creation_date: 2014-11-25T16:35:53Z

**IntAct:EBI-9997581** (This ID present in 3 terms)

> [Term]
> id: GO:1990586
> name: divisome complex
> namespace: cellular_component
> def: "A protein complex required for prokaryotic cell division (FtsZ-dependent cytokinesis). These complexes are assembled and recruited to the cell septum in a strictly controlled sequence and co-ordinate invagination of the cell membrane, inward growth of the peptidoglycan layer, constriction of the outer membrane and separati\
> on of daughter cells." [GOC:bhm, IntAct:EBI-9547109, IntAct:EBI-9997581, PMID:15165235, PMID:21784946]
> is_a: GO:0043234 ! protein complex
> created_by: bhm
> creation_date: 2014-12-15T09:13:38Z
> 
> [Term]
> id: GO:1990587
> name: FtsQBL complex
> namespace: cellular_component
> def: "A protein complex required for prokaryotic cell division (FtsZ-dependent cytokinesis). Part of the divisome. Assembled independently of the other divisome components in the cytoplasm prior to transport to the cell septum. In E. coli consists of FtsB, FtsL and FtsQ." [GOC:bhm, IntAct:EBI-9997581, PMID:15165235, PMID:21784946]
> comment: An example of this is FtsB in E. coli (P0A6S5) in PMID:15165235 (inferred from physical interaction).
> synonym: "FtsB-FtsL-FtsQ complex" EXACT []
> is_a: GO:0044457 ! cell septum part
> is_a: GO:1990586 ! divisome complex
> created_by: bhm
> creation_date: 2014-12-15T09:25:09Z
> 
> [Term]
> id: GO:1990588
> name: FtsBL complex
> namespace: cellular_component
> def: "A protein complex required for prokaryotic cell division (FtsZ-dependent cytokinesis). Part of the divisome. Assembled independently of the other divisome components in the cytoplasm prior to transport to the cell septum. In E. coli consists of FtsB and FtsL." [GOC:bhm, IntAct:EBI-9997581, PMID:15165235, PMID:21784946]
> comment: An example of this is FtsB in E. coli (P0A6S5) in PMID:15165235 (inferred from physical interaction).
> synonym: "FtsB-FtsL complex" EXACT []
> is_a: GO:1990586 ! divisome complex
> created_by: bhm
> creation_date: 2014-12-15T09:33:00Z
>

**Comment by krchristie** [comment]
I also noticed one IntAct reference present in a definition as a comment instead of as a database_cross_reference. If it is decided that these should be general xrefs instead of definition xrefs, I think that will be better done programmatically, but I wanted to fix this one so that it is consistent with the others:  https://github.com/geneontology/go-ontology/pull/14273

**Comment by krchristie** [comment]
These are the two terms I noticed where the IntAct xrefs are listed BOTH as general xrefs and as definition xrefs (Note that in one definition, there is alternative capitalization of "IntACT"):

> id: GO:1990265
> name: platelet-derived growth factor complex
> namespace: cellular_component
> def: "A protein complex consisting of two chains of platelet-derived growth factor (PDGF) subunits. PDGF dimers bind to PDGF receptors in the plasma membrane and induce receptor dimerisation and activation. PDGFs are involved in a wide variety of signalling processes. PDGFs are found in all vertebrates where at least 2 different chains (A and B) exist. In human (and other mammals), four types of PDGF chains (A, B, C, and D) are known which form five different dimers (AA, AB, BB, CC and DD)." [GOC:bhm, IntAct:EBI-2881436, IntAct:EBI-2881443, IntAct:EBI-2881451, PMID:11331882]
> comment: An example of this is PDGFA in human (P04085) in PMID:20534510 (inferred from direct assay).
> xref: IntAct:EBI-2881436
> xref: IntAct:EBI-2881443
> xref: IntAct:EBI-2881451
> created_by: bhm
> creation_date: 2014-01-09T16:09:11Z
> 
> id: GO:1990270
> name: platelet-derived growth factor receptor-ligand complex
> namespace: cellular_component
> def: "A tetrameric protein complex consisting of two platelet-derived growth factor (PDGF) receptor subunits and two PDGF ligand subunits. Binding of the PDGF ligand dimer to the PDGF receptor in the plasma membrane induces receptor dimerisation and activation. PDGFs are involved in a wide variety of signalling processes and are found in all vertebrates. At least two different receptor chains (A and B) and four types of ligand chains (A, B, C, and D) are known forming a wide variety of combinations of receptor-ligand complexes." [GOC:bhm, IntACT:EBI-9080360, PMID:11331882]
> xref: IntAct:EBI-9080360
> is_a: GO:0043234 ! protein complex
> created_by: bhm
> creation_date: 2014-01-14T11:12:22Z
>

**Comment by krchristie** [comment]
Tagging @ukemi @cmungall @hdrabkin for input on where IntAct IDs should appropriately be referenced. Currently there is a mix of a few as general xrefs similarly to where things like EC numbers are referenced, and the majority as definition xrefs.

There are also a few IntAct IDs that are listed for more than one GO term (listed above in a previous long comment). I know we're trying to avoid this for EC numbers, so I've asked @bmeldal to comment on whether this is really intended.

Also worth mentioning that @bmeldal indicated that

> There should really be nearly 2000 xrefs as almost all complexes in the CP are linked to a complex term in GO. 

so it might also be worth discussing some automated way to fill all of these.

**Comment by hdrabkin** [comment]
For any of the GO complex terms I've done I don't recall there being any xref to a CP complex at all.

**Comment by ukemi** [comment]
The Intact complex references should go in the term xrefs, not the definition xrefs.

**Comment by bmeldal** [comment]
Thank you @krchristie for raising the duplicated xref issue. These are copy/paste mistakes from when I requested sets of parent and child terms and in some cases siblings and didn't swap the IDs consistently...

Please correct as follows:

GO:1990662:
Should have EBI-10099500 & EBI-10099571 as general xrefs. Delete the existing ones.

GO:1990658:
remove IntAct xrefs - this is a general grouping class and the xrefs are more appropriate for the child term.

GO:1990351:
remove IntAct xrefs - this is a general grouping class and the xrefs are more appropriate for the child term.

GO:1990563:
remove IntAct xrefs - this is a general grouping class and the xrefs are more appropriate for the child term.

GO:1990586:
remove IntAct xrefs - this is a general grouping class and the xrefs are more appropriate for the child term.

GO:1990587:
add EBI-9547109 as general xref and removed EBI-9997581 (belongs to GO:1990588 only)

GO:1990265 & GO:1990270:
I guess these duplicates got fixed by @cmungall 's script yesterday?
DB names should always be "IntAct" (well, soon will be "Complex Portal"!)

Thanks you, all!
Birgit

**Comment by krchristie** [comment]
Thanks for going through these Birgit and providing the appropriate corrections. That's what I suspected had happened. I've corrected them in as indicated, either to remove from grouping terms or to replace with correct IDs for specific terms.

With respect to these last two: GO:1990265 & GO:1990270:

> I guess these duplicates got fixed by @cmungall 's script yesterday?
> DB names should always be "IntAct" (well, soon will be "Complex Portal"!)

Chris's script did remove all of the definition dbxrefs that used the correct capitalization of "IntAct", but did not remove the incorrect one of "IntACT" from GO:1990270, so I have now removed that as well.
